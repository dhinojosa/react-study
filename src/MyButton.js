import * as React from "react";

class MyButton extends React.Component {
    constructor(props){
        super(props);
        this.state = {counter: 1};
    }

    clickEvent = () => {
        this.setState((prevState) => ({
            counter: prevState.counter + 1
        }));
    };

    render() {
        return <button onClick={this.clickEvent}>{this.state.counter}</button>
    }
}

export default MyButton