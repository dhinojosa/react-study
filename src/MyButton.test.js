import React from 'react';
import MyButton from "./MyButton"
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


it ('should be able to render', () => {
    const wrapper = shallow(<MyButton/>);
    wrapper.find("button").simulate('click');
    wrapper.find("button").simulate('click');
    wrapper.find("button").simulate('click');
    const result = wrapper.find("button").props();
    expect(result.children).toBe(4);
});


it('should be able to keep track of the dom with each subsequent call', () => {
    const component = renderer.create(
        <MyButton/>
    );

    let button = component.toJSON();
    expect(button).toMatchSnapshot();

    button.props.onClick();
    expect(button).toMatchSnapshot();

    button.props.onClick();
    expect(button).toMatchSnapshot();
});