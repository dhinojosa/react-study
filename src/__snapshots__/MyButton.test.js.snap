// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be able to keep track of the dom with each subsequent call 1`] = `
<button
  onClick={[Function]}
>
  1
</button>
`;

exports[`should be able to keep track of the dom with each subsequent call 2`] = `
<button
  onClick={[Function]}
>
  1
</button>
`;

exports[`should be able to keep track of the dom with each subsequent call 3`] = `
<button
  onClick={[Function]}
>
  1
</button>
`;
